### PR TITLE
Add automated Python linting using flake8 and GitHub Actions

### PR DESCRIPTION
**Pull request by a-gavin**
This pull request adds automated Python script linting to run on all push and pull requests to the LANforge scripts repository. It also adds support for local runs of the same linting checks using the `pre-commit` tool. More details and local setup instructions for the `pre-commit` tool are listed in the `CONTRIBUTING.md` file introduced in this pull request.

An example run of this linting check can be found [here](https://github.com/a-gavin/lanforge-scripts/actions/runs/11040441885) in my fork.

The goal with this is to ensure consistent code quality moving forward as best as possible. By using automated linting through GitHub Actions, this should also ease the burden on reviewers when reviewing future submissions. New scripts will be required to pass linting checks and possibly future submissions will involve pull requests which block based on linting checks. Details are subject to discussion.

Given the many existing Python scripts in the project that will not meet linting standards, this pull requests configures linting to opt-out most of the existing scripts and only check a selected few which meet the linting standards already. The opt-out is done with the `# flake8: noqa` tag at the top of the Python file. The `CONTRIBUTING.md` file introduced in this pull request documents this as well.

With this patch series, the following files are checked:
- All `lanforge_client/examples/` scripts
- `py-scripts/create_bridge.py`
- `py-scripts/create_macvlan.py`
- `py-scripts/create_qvlan.py`
- `py-scripts/create_station.py`

### CODE REVIEW

**Comment by a-gavin** [comment]
Force pushed just now to comment out bugbear in the `pre-commit` configuration as well. Only the `pre-commit` introduction commit and the commit after were updated, as you can see in the timeline above.

File: py-scripts/lf_mixed_traffic.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
-#!/usr/bin/env python3
+#! usr/bin/env python3
```
Comment (memnochproxy) [inline py-scripts/lf_mixed_traffic.py:1]: Typo in shebang

File: py-scripts/lf_json_util.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-
```
Comment (memnochproxy) [inline py-scripts/lf_json_util.py:1]: No header at all in this file?

File: label-printer/label-printer.py
Comment (memnochproxy) [inline label-printer/label-printer.py:1]: I thought you wanted to pull this out of the lanforge-scripts repo and put it in btbits/tools?

File: archive/rssi_validation/src/processing.py [hunk 3]
```diff
@@ -1,4 +1,5 @@
```
Comment (memnochproxy) [inline archive/rssi_validation/src/processing.py:1]: no shebang?

File: lanforge_client/strutil.py [hunk 4]
```diff
@@ -1,3 +1,4 @@
+# flake8: noqa
```
Comment (memnochproxy) [inline lanforge_client/strutil.py:1]: I think things without a shebang probably need a statement saying *required by foo.py*

File: py-json/LANforge/add_file_endp.py [hunk 5]
```diff
@@ -1,3 +1,4 @@
+# flake8: noqa
```
Comment (memnochproxy) [inline py-json/LANforge/add_file_endp.py:1]: wants a *required by*

File: py-json/LANforge/add_vap.py [hunk 6]
```diff
@@ -1,3 +1,4 @@
+# flake8: noqa
```
Comment (memnochproxy) [inline py-json/LANforge/add_vap.py:1]: wants a *required by*

File: py-json/LANforge/set_port.py [hunk 7]
```diff
@@ -1,3 +1,4 @@
+# flake8: noqa
```
Comment (memnochproxy) [inline py-json/LANforge/set_port.py:1]: wants a *required by*

File: py-json/LANforge/set_wifi_radio.py [hunk 8]
```diff
@@ -1,3 +1,4 @@
+# flake8: noqa
```
Comment (memnochproxy) [inline py-json/LANforge/set_wifi_radio.py:1]: wants a *required by*

File: py-json/cv_test_reports.py [hunk 9]
```diff
@@ -1,3 +1,4 @@
+# flake8: noqa
```
Comment (memnochproxy) [inline py-json/cv_test_reports.py:1]: wants a *required by*

File: py-scripts/scripts_deprecated/tip-cicd-sanity/Throughput_Test.py [hunk 10]
```diff
@@ -1,3 +1,4 @@
+# flake8: noqa
```
Comment (memnochproxy) [inline py-scripts/scripts_deprecated/tip-cicd-sanity/Throughput_Test.py:1]: wants *required by* or shebang

File: py-scripts/sandbox/cc_module_opensync.py [hunk 11]
```diff
@@ -1,3 +1,4 @@
+# flake8: noqa
```
Comment (memnochproxy) [inline py-scripts/sandbox/cc_module_opensync.py:1]: missing shebang

**Review by memnochproxy — CHANGES_REQUESTED** [review]
I think now is a good time to get at least a few of these scripts with a comment or shebang at the top of the file.

File: py-scripts/lf_mixed_traffic.py (hunk 1 above)
Comment (a-gavin) [reply py-scripts/lf_mixed_traffic.py:1]: Will address in follow-up

File: archive/rssi_validation/src/processing.py (hunk 3 above)
Comment (a-gavin) [reply archive/rssi_validation/src/processing.py:1]: This script is archived, but I can follow-up and add a shebang if desired. Definitely want to do that for actively-used scripts, though

File: label-printer/label-printer.py
Comment (a-gavin) [reply label-printer/label-printer.py:1]: Snuck in there. Will follow-up in this PR w/ removal

File: lanforge_client/strutil.py (hunk 4 above)
Comment (a-gavin) [reply lanforge_client/strutil.py:1]: Agreed. Can do that in a follow-up change (outside of this PR)

File: py-json/LANforge/add_file_endp.py (hunk 5 above)
Comment (a-gavin) [reply py-json/LANforge/add_file_endp.py:1]: See `strutil.py` comment.

File: py-json/LANforge/add_vap.py (hunk 6 above)
Comment (a-gavin) [reply py-json/LANforge/add_vap.py:1]: See `strutil.py` comment.

File: py-json/LANforge/set_port.py (hunk 7 above)
Comment (a-gavin) [reply py-json/LANforge/set_port.py:1]: See `strutil.py` comment.

File: py-json/LANforge/set_wifi_radio.py (hunk 8 above)
Comment (a-gavin) [reply py-json/LANforge/set_wifi_radio.py:1]: See `strutil.py` comment.

File: py-json/cv_test_reports.py (hunk 9 above)
Comment (a-gavin) [reply py-json/cv_test_reports.py:1]: See `strutil.py` comment.

File: py-scripts/lf_json_util.py (hunk 2 above)
Comment (a-gavin) [reply py-scripts/lf_json_util.py:1]: Guess not. Can add to the list for follow-up canges (outside of this PR).

File: py-scripts/sandbox/cc_module_opensync.py (hunk 11 above)
Comment (a-gavin) [reply py-scripts/sandbox/cc_module_opensync.py:1]: This script is in sandbox, but I can add a shebang if desired. This seems outside the scope of this PR, though

File: py-scripts/scripts_deprecated/tip-cicd-sanity/Throughput_Test.py (hunk 10 above)
Comment (a-gavin) [reply py-scripts/scripts_deprecated/tip-cicd-sanity/Throughput_Test.py:1]: This script is deprecated. I can add this, but unlikely needed given that it's on its way out.

File: py-scripts/lf_mixed_traffic.py (hunk 1 above)
Comment (a-gavin) [reply py-scripts/lf_mixed_traffic.py]: Updated my branch to address this oversight.

File: archive/rssi_validation/src/processing.py (hunk 3 above)
Comment (memnochproxy) [reply archive/rssi_validation/src/processing.py:1]: archive doesn't matter

File: py-scripts/scripts_deprecated/tip-cicd-sanity/Throughput_Test.py (hunk 10 above)
Comment (memnochproxy) [reply py-scripts/scripts_deprecated/tip-cicd-sanity/Throughput_Test.py:1]: deprecated, yeah, don't care

File: py-scripts/sandbox/cc_module_opensync.py (hunk 11 above)
Comment (memnochproxy) [reply py-scripts/sandbox/cc_module_opensync.py:1]: sandbox, yeah don't care

**Review by memnochproxy — APPROVED** [review]

File: py-scripts/lf_mixed_traffic.py
Comment (smileyrekiere) [inline py-scripts/lf_mixed_traffic.py:1]: Shouldn't this be   #!/usr/bin/env python3  

locally modified to be   #!usr/bin/env python3 and get

user@user:~/git/lanforge-scripts/py-scripts$ ./lf_mixed_traffic.py --help
bash: ./lf_mixed_traffic.py: usr/bin/env: bad interpreter: No such file or directory

Comment (a-gavin) [reply py-scripts/lf_mixed_traffic.py:1]: Ah that was sloppy of me. I didn't double check my fix. Let me fix this properly and update the PR.

Comment (smileyrekiere) [reply py-scripts/lf_mixed_traffic.py:1]: Thank-you

**Review by smileyrekiere — APPROVED** [review]

**Pull request merged**